plugins/net_plugin/net_plugin.cpp

```diff
@@ -2841,18 +2842,24 @@ namespace eosio {
                if(check.get() == this)
                   continue;
                if(check->connected() && check->peer_name() == msg.p2p_address) {
-                  // It's possible that both peers could arrive here at relatively the same time, so
-                  // we need to avoid the case where they would both tell a different connection to go away.
-                  // Using the sum of the initial handshake times of the two connections, we will
-                  // arbitrarily (but consistently between the two peers) keep one of them.
-                  std::unique_lock<std::mutex> g_check_conn( check->conn_mtx );
-                  auto check_time = check->last_handshake_sent.time + check->last_handshake_recv.time;
-                  g_check_conn.unlock();
-                  g_conn.lock();
-                  auto c_time = last_handshake_sent.time;
-                  g_conn.unlock();
-                  if (msg.time + c_time <= check_time)
-                     continue;
+                  if (net_version < dup_goaway_resolution || msg.network_version < dup_goaway_resolution) {
+                     // It's possible that both peers could arrive here at relatively the same time, so
+                     // we need to avoid the case where they would both tell a different connection to go away.
+                     // Using the sum of the initial handshake times of the two connections, we will
+                     // arbitrarily (but consistently between the two peers) keep one of them.
+                  
+                     std::unique_lock<std::mutex> g_check_conn( check->conn_mtx );
+                     auto check_time = check->last_handshake_sent.time + check->last_handshake_recv.time;
+                     g_check_conn.unlock();
+                     g_conn.lock();
+                     auto c_time = last_handshake_sent.time;
+                     g_conn.unlock();
+                     if (msg.time + c_time <= check_time)
+                        continue;
+                  }
+                  else if (my_impl->p2p_address < msg.p2p_address) {
+                     continue; 
+                  }
```

Comment:
I'd like a comment like "only the p2p node with the higher of the 2 addresses will decide which connection is a duplicate, so there is no chance for both connections to be closed"